### PR TITLE
[refactor] Vault headers

### PR DESCRIPTION
**Pull request by garguelles**
## Checklist (Mandatory):

- [x] I have named the PR correctly
- [x] I have tested new code locally
- [x] I have done a smoke test of the larger feature locally
- [x] I have added at least one reviewer
- [x] I have assigned the PR to myself
- [x] I have merged develop (parent) branch into my feature branch and resolved all conflicts
- [x] I have ensured any new UI changes comply with UX/UI designers 

## Checklist (Optional):

- [ ] I have added necessary documentation, including instructions how to test
- [ ] I have written the code according to the adopted style guide and linter doesn't fail
- [ ] I have confirmed no errors were introduced in the developer console with the PR

## Additional notes

```
What does this implement/fix? Explain your changes.
- Implement new design for vault headers
- Extract vaultHeader as molecule component
- TokenDetail atom component
- Infobox atom component
Are there any scripts that need running (eg. yarn install)?
Are there any edge cases that need checking?
Are there any specific test instructions?
Are there any Loom videos or screenshot that can be linked explaining the changes better?
```

### CODE REVIEW

File: src/components/atoms/InfoBox/infoBox.atom.types.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export interface IInfoBox {
+  usdValue?: string;
```
Comment (pajicf) [inline src/components/atoms/InfoBox/infoBox.atom.types.ts:2]: Why not treat this as a generic header? That way it's not strictly binded to a specific domain value but can be used more generically

File: src/components/atoms/InfoBox/infoBox.atom.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import {EColor, EFontWeight} from "../../../shared/types/styles.types";
+import Skeleton from "../Skeleton/skeleton.atom";
+import Typography from "../Typography/typography.atom";
+import { IInfoBox } from "./infoBox.atom.types";
+import styles from "./infoBox.atom.module.scss";
+
+const InfoBox: React.FC<IInfoBox> = (props) => {
+  const {usdValue, value, footer, loading = false} = props;
+  // temp - find a better way to handle this
+  const showUsdValue = usdValue && usdValue !== "$0";
```
Comment (pajicf) [inline src/components/atoms/InfoBox/infoBox.atom.tsx:11]: I think this is on parent component to decide if it wants to pass something to the optional prop or not. If something is passed we just render it here, otherwise no render

File: src/components/atoms/Link/link.atom.types.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 export interface ILinkProps {
   link: string;
   newTab?: boolean;
+  className?: string;
```
Comment (pajicf) [inline src/components/atoms/Link/link.atom.types.ts:4]: We can use`IClassableComponent` util.type here

File: src/shared/constants/common.constants.ts [hunk 4]
```diff
@@ -17,6 +17,23 @@ export const supportedLocaleList: ESupportedLocales[] = [
   ESupportedLocales.INTERSLAVIC
 ];
 
+export enum EAssetType {
```
Comment (pajicf) [inline src/shared/constants/common.constants.ts:20]: Enums shouldn't be really be used as a way to record "const" data such as labels as here. They ideally just classify something of what type it is from some constrained set of options.

In that case, we would move this to a .type file such as metadata.types.ts

Another reason for the EAssetType and AssetTypeLabel being needed as a separate const is that this is a string which also need to be in the i18n form.

File: src/components/atoms/TokenDetail/tokenDetail.atom.module.scss [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+
+    .purchaseLink {
+      font-size: 13px;
+      font-weight: 400;
+      line-height: 16px;
+      text-decoration: underline;
+      color: map-get($colors, "white");
```
Comment (pajicf) [inline src/components/atoms/TokenDetail/tokenDetail.atom.module.scss:23]: We should also add onHover effect to the link

File: src/components/atoms/TokenDetail/tokenDetail.atom.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        <Logo className={styles.logo} /> 
+      </div>
+      <div className={styles.tokenLabel}>
+        <Typography
+          fontSize={16}
+          fontWeight={EFontWeight.BOLD}
+        >
+          {name}
+        </Typography>
+        {purchaseLink && <Link className={styles.purchaseLink} link={purchaseLink}>{assetType}</Link>}
```
Comment (pajicf) [inline src/components/atoms/TokenDetail/tokenDetail.atom.tsx:23]: We should stop click propagation when click on the link so that the dropdown doesn't open. We can do this inside the Link component with onClick listener maybe?

File: src/components/molecules/VaultHeader/vaultHeader.molecule.module.scss [hunk 7]
```diff
@@ -0,0 +1,16 @@
+.vaultHeader {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  cursor: pointer;
+  height: 107px;
+
+  .tokenInfoColumn {
+    flex-grow: 1;
+    //max-width: 339px;
```
Comment (pajicf) [inline src/components/molecules/VaultHeader/vaultHeader.molecule.module.scss:10]: Please remove obsolete/unused code

File: src/components/molecules/VaultHeader/vaultHeader.molecule.tsx [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+          name={getTokenLabel()!}
+          purchaseLink={buyTokenUrl}
+          logo={tokenLogo}
+        />
+      </div>
+      <div className={styles.tokenDataColumn}>
+        <InfoBox 
+          usdValue={getFormattedBalanceInUSD()} 
+          value={getFormattedBalance()} 
+          footer="Wallet"
```
Comment (pajicf) [inline src/components/molecules/VaultHeader/vaultHeader.molecule.tsx:103]: i18n

File: src/components/molecules/VaultHeader/vaultHeader.molecule.tsx [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+          value={getFormattedBalance()} 
+          footer="Wallet" 
+          loading={loading}
+        />
+      </div>
+      <div className={styles.tokenDataColumn}>
+        <InfoBox 
+          usdValue={getDepositedValueInUSD()}
+          value={getFormattedUserShares()} 
+          footer="Deposited"
```
Comment (pajicf) [inline src/components/molecules/VaultHeader/vaultHeader.molecule.tsx:111]: i18n

File: src/components/molecules/VaultHeader/vaultHeader.molecule.tsx [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+          usdValue={getDepositedValueInUSD()}
+          value={getFormattedUserShares()} 
+          footer="Deposited" 
+          loading={loading}
+        />
+      </div>
+      <div className={styles.tokenDataColumn}>
+        <InfoBox 
+          value={`${apy}%`} 
+          footer="APY"
```
Comment (pajicf) [inline src/components/molecules/VaultHeader/vaultHeader.molecule.tsx:118]: i18n

**Review by pajicf — CHANGES_REQUESTED** [review]

File: src/components/atoms/InfoBox/infoBox.atom.tsx (hunk 2 above)
Comment (garguelles) [reply src/components/atoms/InfoBox/infoBox.atom.tsx:11]: makes sense. Will add a showHeader flag on the interface

File: src/shared/constants/web3.constants.ts [hunk 11]
```diff
@@ -120,13 +120,29 @@ export const tokenLabels: ValueByTokenAndNetwork = {
 
 // @TODO Abstract a bit more
 export const buyTokenUrlByTokenAndNetwork: ValueByTokenAndNetwork = {
```
Comment (pajicf) [inline src/shared/constants/web3.constants.ts:122]: Lots of string repetition here. If just the domain changed for Quickswap, we have 7 changes Maybe we can create util function in web3.util?
Something like `getDexUrl(ESupportedToken, Network)` and call it here in the map

**Review by pajicf — CHANGES_REQUESTED** [review]
Requested changes for the new additions. All else resolved, good job!

File: src/shared/constants/web3.constants.ts (hunk 11 above)
Comment (garguelles) [reply src/shared/constants/web3.constants.ts]: So instead of calling the getDexUrl in the object map. I just completely abstracted the dexUrl generation in that method since calling it in the constants won't work due to cyclic dependency. Handling will depend on the asset type. LP will still have an object map since each LP might come from different dexes. Tokens will just have options depending on the EChainId

**Review by pajicf — APPROVED** [review]
